### DISCUSSION TITLE
Literal types showing up as string with no value in OpenAPI output.

### DISCUSSION THREAD

**Discussion by jjwwiodyne** [fastapi/fastapi#2955] [Questions] [answered]
fastapi: 0.62.0
pydantic: 1.7.3

When using a pydantic model that has a Literal field, the type is shown in `openapi.json` as `str` without also showing its required value:

```
class SomeModel(BaseModel):
    event_type:Literal['SomeModelType']
```

openapi.json:
```
               "event_type":{
                  "title":"Event Type",
                  "type":"string",
                  "description":"Event type."
               },
```

How can I get the OpenAPI output to show the literal value so callers know what it has to be?

**Comment by Mause** [comment]
You could probably just use an enum?

**Comment by jjwwiodyne** [comment]
Using a Union of models with a descriminator on the event type field. If you use an enum than each unioned model shows the same list of allowed types instead of one type.

**Reply by Mause** [reply]
Yes, one enum per model variant. Not very efficient but it would work

**Comment by jjwwiodyne** [comment]
That would be difficult to maintain just for documentation. It would be better if Literals were treated as constants instead of variable strings. 

Sent via iPhone

**Comment by jjwwiodyne** [comment, marked as answer]
This is fixed in Pydantic 1.8.1: https://github.com/samuelcolvin/pydantic/pull/2348
